src/code.cloudfoundry.org/cf-tcp-router/config/config.go

```diff
@@ -120,71 +127,42 @@ func (c *Config) initConfigFromFile(path string, enableCertCreation bool) error
 		c.DrainWaitDuration = DrainWaitDefault
 	}
 
-	if enableCertCreation && len(c.FrontendTLSJob) > 0 {
-		owner, err := user.Lookup("root")
-		if err != nil {
-			return err
+	for i, cert := range c.FrontendTLSJob {
+		// validations
+		name := strings.TrimSpace(cert.Name)
+		certChain := strings.TrimSpace(cert.CertChain)
+		privateKey := strings.TrimSpace(cert.PrivateKey)
+		if name == "" || privateKey == "" || certChain == "" {
+			return fmt.Errorf("frontend_tls[%d]: invalid name, cert_chain, or private_key", i)
```

Comment:
This error message should be improved:

1. the error says "invalid", but all this actually tests for is if the items are missing.
2. the error does not specify WHICH item is the issue. Please create a different error message for each item.

Reply:
👍 However, just like to note that I do see tests for all these of these error cases.